### ISSUE TITLE
Dataset format

### ISSUE THREAD

**Issue by dXDb** [bearcatt/LaBERT#5]
Hi!
Thank you for your last reply.

I have a question about dataset. I did download this [link](https://github.com/LuoweiZhou/VLP#-data-preparation)'s coco data part1 and part2 and exchanged files name coco_detection_vg_100dets_gvd_checkpoint_trainval_cls(0to999).h5 -> cls(0to999).h5, coco_detection_vg_100dets_gvd_checkpoint_trainval_feat(0to999).h5 -> feat(0to999).h5, and coco_detection_vg_thresh0.2_feat_gvd_checkpoint_trainvaltest.h5 -> region_bbox.h5.
But an error has occurred. (command at terminal : python -m torch.distributed.launch --nproc_per_node=1 --master_port=4396 train.py save_dir result/  samples_per_gpu 1)
![image](https://user-images.githubusercontent.com/51734430/97212298-8af1c100-1803-11eb-8417-ac121e4b3d3e.png)

I add source code train.py at 90 line and dataset.py at 86line so these have prints.
train.py
* print(pred_scores, gt_token_ids)
loss = criterion(pred_scores, gt_token_ids)

dataset.py
* print(name)
with h5py.File(osp.join(self.root, f'feat{name[-3:]}.h5'), 'r') as features, \
        h5py.File(osp.join(self.root, f'cls{name[-3:]}.h5'), 'r') as classes, \
        h5py.File(osp.join(self.root, f'region_bbox.h5'), 'r') as bboxes:

Is it right to set up the dataset like this? If not, could you explain the dataset guidelines in more detail?
Thank you for providing good model.   :)

**Comment by dXDb** [comment]
I used another command and it worked. But is it right to set the dataset like that?
Looking at the loss, it seems that there is no learning at all.

before : python -m torch.distributed.launch --nproc_per_node=1 --master_port=4396 train.py save_dir result/ samples_per_gpu 1
after : python -m torch.distributed.launch --nproc_per_node=1 --master_port=4396 train.py save_dir result/ samples_per_gpu 64

train_log
2020-10-27 03:48:39,165 train INFO: iter: 20  time: 0.6309  mem: 13.97  lr: 0.00000100  loss: 7.3856
2020-10-27 03:48:51,644 train INFO: iter: 40  time: 0.6122  mem: 13.97  lr: 0.00000200  loss: 6.6550
2020-10-27 03:49:04,011 train INFO: iter: 60  time: 0.5900  mem: 13.97  lr: 0.00000300  loss: 6.0712
...
2020-10-27 04:14:44,868 train INFO: iter: 2520  time: 0.6463  mem: 13.97  lr: 0.00004997  loss: 4.4873
2020-10-27 04:14:57,367 train INFO: iter: 2540  time: 0.6409  mem: 13.97  lr: 0.00004997  loss: 4.2791
2020-10-27 04:15:09,790 train INFO: iter: 2560  time: 0.6345  mem: 13.97  lr: 0.00004997  loss: 3.7747
...
2020-10-27 04:40:56,327 train INFO: iter: 5020  time: 0.6104  mem: 13.98  lr: 0.00004980  loss: 3.6449
2020-10-27 04:41:08,772 train INFO: iter: 5040  time: 0.6091  mem: 13.98  lr: 0.00004979  loss: 3.7078
2020-10-27 04:41:21,316 train INFO: iter: 5060  time: 0.6115  mem: 13.98  lr: 0.00004979  loss: 3.8532
...
2020-10-27 05:06:58,271 train INFO: iter: 7520  time: 0.6346  mem: 13.98  lr: 0.00004947  loss: 3.8869
2020-10-27 05:07:10,832 train INFO: iter: 7540  time: 0.6064  mem: 13.98  lr: 0.00004946  loss: 3.4876
2020-10-27 05:07:23,233 train INFO: iter: 7560  time: 0.6268  mem: 13.98  lr: 0.00004946  loss: 4.0862
...
2020-10-27 05:32:55,560 train INFO: iter: 10020  time: 0.6443  mem: 13.98  lr: 0.00004898  loss: 3.7881
2020-10-27 05:33:07,952 train INFO: iter: 10040  time: 0.6405  mem: 13.98  lr: 0.00004898  loss: 3.5467
2020-10-27 05:33:20,421 train INFO: iter: 10060  time: 0.6030  mem: 13.98  lr: 0.00004897  loss: 3.4687
...

**Comment by bearcatt** [comment]
Hi, sorry for the late response. I think your train log is normal. If you keep training the model, the loss will finally decrease to 2~3. Moreover, I use a batch size of 256 in the paper with 4 GPU cards to obtain the reported performance. 

As for the dataset guidelines, I'm a little busy recently and could not provide you the guideline in the next two weeks, sorry for the inconvenience incurred. Before that, you may keep training your model to see if the performance got improved.

**Comment by dXDb** [comment]
Thank you for reply! I will keep more training.
Before that, can you take a look at my eval log?
Is it right to continue learning in this state?

My GPU is only one TITAN RTX

### with your generator.pth
2020-10-27 11:12:02,119 inference INFO: Running with config:
boundaries: ((7, 9), (10, 14), (15, 19), (20, 25))
checkpoint_time: 2500
data_dir: data
device: cuda
distributed: True
infer:
  eos_decay: (1.0, 0.88, 0.95, 1.0)
  steps: (10, 15, 20, 20)
log_time: 20
loss:
  balance_weight: 0.5
  label_smoothing: 0.1
model_path: pretrained/generator.pth
num_workers: 3
pretrained_bert: pretrained/bert.pth
samples_per_gpu: 64
save_dir: output
scheduler:
  max_steps: 100000
  warmup_steps: 1000
solver:
  betas: (0.9, 0.999)
  grad_clip: 1.0
  lr: 5e-05
  weight_decay: 0.01
2020-10-27 11:12:03,648 inference INFO: Loading checkpoint from pretrained/generator.pth
2020-10-27 11:12:21,853 inference INFO: Start inferencing
2020-10-27 11:28:34,301 inference INFO: batch_time: 12.3094 batch_memory: 3.63
2020-10-27 11:28:34,302 inference INFO: Saving results to output/caption_results.json
2020-10-28 08:50:29,703 evaluate INFO: loading ground-truths from data/id2captions_test.json
2020-10-28 09:13:01,811 evaluate INFO: loading ground-truths from data/id2captions_test.json
2020-10-28 09:13:02,285 evaluate INFO: loading predictions from output/caption_results.json
2020-10-28 09:13:03,415 evaluate INFO: Start evaluating
2020-10-28 09:20:20,353 evaluate INFO: loading ground-truths from data/id2captions_test.json
2020-10-28 09:20:20,764 evaluate INFO: loading predictions from output/caption_results.json
2020-10-28 09:20:21,700 evaluate INFO: Start evaluating
2020-10-28 09:20:34,242 evaluate INFO: C: 1.0157 S: 0.1947 M: 0.2543 R: 0.5464 B1: 0.7249 B2: 0.5631 B3: 0.4171 B4: 0.3001
2020-10-28 09:21:03,046 evaluate INFO: C: 1.1817 S: 0.2182 M: 0.2837 R: 0.5743 B1: 0.7759 B2: 0.6131 B3: 0.4688 B4: 0.3534
2020-10-28 09:21:46,994 evaluate INFO: C: 0.9049 S: 0.2230 M: 0.2856 R: 0.5307 B1: 0.6678 B2: 0.5058 B3: 0.3702 B4: 0.2678
2020-10-28 09:22:55,635 evaluate INFO: C: 0.3985 S: 0.2235 M: 0.2775 R: 0.4696 B1: 0.5607 B2: 0.4090 B3: 0.2862 B4: 0.1977
2020-10-28 09:22:55,636 evaluate INFO: 4 level ensemble CIDEr: 1.3764  3 level ensemble CIDEr: 1.3734  2 level ensemble CIDEr: 1.3149


### with my generator.pth(near 200K iters pth files)
2020-10-29 09:15:19,588 inference INFO: Running with config:
boundaries: ((7, 9), (10, 14), (15, 19), (20, 25))
checkpoint_time: 2500
data_dir: data
device: cuda
distributed: True
infer:
  eos_decay: (1.0, 0.88, 0.95, 1.0)
  steps: (10, 15, 20, 20)
log_time: 20
loss:
  balance_weight: 0.5
  label_smoothing: 0.1
model_path: result/MinLoss_2.4397.pth
num_workers: 3
pretrained_bert: pretrained/bert.pth
samples_per_gpu: 64
save_dir: output
scheduler:
  max_steps: 400000
  warmup_steps: 1000
solver:
  betas: (0.9, 0.999)
  grad_clip: 1.0
  lr: 5e-05
  weight_decay: 0.01
2020-10-29 09:15:20,976 inference INFO: No checkpoint found.
2020-10-29 09:15:29,067 inference INFO: Start inferencing
2020-10-29 09:31:30,793 inference INFO: batch_time: 12.1737 batch_memory: 3.63
2020-10-29 09:31:30,793 inference INFO: Saving results to output/caption_results.json
2020-10-29 09:34:08,544 evaluate INFO: loading ground-truths from data/id2captions_test.json
2020-10-29 09:34:09,008 evaluate INFO: loading predictions from output/caption_results.json
2020-10-29 09:34:10,078 evaluate INFO: Start evaluating
2020-10-29 09:34:39,491 evaluate INFO: C: 0.0001 S: 0.0004 M: 0.0050 R: 0.0002 B1: 0.0002 B2: 0.0000 B3: 0.0000 B4: 0.0000
2020-10-29 09:35:19,293 evaluate INFO: C: 0.0002 S: 0.0004 M: 0.0044 R: 0.0005 B1: 0.0004 B2: 0.0000 B3: 0.0000 B4: 0.0000
2020-10-29 09:36:41,027 evaluate INFO: C: 0.0001 S: 0.0006 M: 0.0046 R: 0.0003 B1: 0.0002 B2: 0.0000 B3: 0.0000 B4: 0.0000
2020-10-29 09:38:35,327 evaluate INFO: C: 0.0000 S: 0.0013 M: 0.0082 R: 0.0002 B1: 0.0001 B2: 0.0000 B3: 0.0000 B4: 0.0000
2020-10-29 09:38:35,329 evaluate INFO: 4 level ensemble CIDEr: 0.0004  3 level ensemble CIDEr: 0.0003  2 level ensemble CIDEr: 0.0003
2020-10-29 09:53:14,716 inference INFO: Running with config:
boundaries: ((7, 9), (10, 14), (15, 19), (20, 25))
checkpoint_time: 2500
data_dir: data
device: cuda
distributed: True
infer:
  eos_decay: (1.0, 0.88, 0.95, 1.0)
  steps: (10, 15, 20, 20)
log_time: 20
loss:
  balance_weight: 0.5
  label_smoothing: 0.1
model_path: result/MinLoss_2.6409.pth
num_workers: 3
pretrained_bert: pretrained/bert.pth
samples_per_gpu: 64
save_dir: output
scheduler:
  max_steps: 200000
  warmup_steps: 1000
solver:
  betas: (0.9, 0.999)
  grad_clip: 1.0
  lr: 5e-05
  weight_decay: 0.01
2020-10-29 09:53:16,108 inference INFO: No checkpoint found.
2020-10-29 09:53:24,267 inference INFO: Start inferencing
2020-10-29 10:09:26,326 inference INFO: batch_time: 12.1779 batch_memory: 3.63
2020-10-29 10:09:26,326 inference INFO: Saving results to output/caption_results.json
2020-10-29 10:40:25,891 evaluate INFO: loading ground-truths from data/id2captions_test.json
2020-10-29 10:40:26,304 evaluate INFO: loading predictions from output/caption_results.json
2020-10-29 10:40:27,394 evaluate INFO: Start evaluating
2020-10-29 10:40:58,027 evaluate INFO: C: 0.0002 S: 0.0003 M: 0.0060 R: 0.0004 B1: 0.0004 B2: 0.0000 B3: 0.0000 B4: 0.0000
2020-10-29 10:41:40,232 evaluate INFO: C: 0.0001 S: 0.0007 M: 0.0058 R: 0.0006 B1: 0.0006 B2: 0.0000 B3: 0.0000 B4: 0.0000
2020-10-29 10:43:03,510 evaluate INFO: C: 0.0001 S: 0.0002 M: 0.0030 R: 0.0004 B1: 0.0003 B2: 0.0000 B3: 0.0000 B4: 0.0000
2020-10-29 10:44:21,415 evaluate INFO: C: 0.0000 S: 0.0000 M: 0.0012 R: 0.0003 B1: 0.0002 B2: 0.0000 B3: 0.0000 B4: 0.0000
2020-10-29 10:44:21,415 evaluate INFO: 4 level ensemble CIDEr: 0.0004  3 level ensemble CIDEr: 0.0004  2 level ensemble CIDEr: 0.0003

**Comment by dXDb** [comment]
My pth path is wrong... sorry... My pth path is result/train/*.pth but I wrote result/*.pth.
Thank you for reply always!
